compiler-plugin/src/main/java/io/ballerina/lib/cdc/compiler/validator/CdcFunctionValidator.java

```diff
@@ -0,0 +1,324 @@
+                if (actualMemberType.typeKind() != RECORD) {
+                    reportErrorDiagnostics(INVALID_PARAM_TYPE, requiredParam.location(),
+                            requiredParam.paramName().map(Node::toString).orElse(""), RECORD.getName());
+                    return false;
+                }
+            }
+        } else if (actualTypeDesc != RECORD) {
+            reportErrorDiagnostics(INVALID_PARAM_TYPE, requiredParam.typeName().location(),
+                    requiredParam.paramName().map(Node::toString).orElse(""), RECORD.getName());
+            return false;
+        }
+        return true;
+    }
+
+    private void validateStringParameter(ParameterNode parameterNode) {
+        if (!(parameterNode instanceof RequiredParameterNode requiredParam)) {
+            reportDiagnosticsForNonRequiredParam(parameterNode);
+            return;
+        }
+
+        Optional<Symbol> typeSymbolOpt = this.semanticModel.symbol(requiredParam);
+        if (typeSymbolOpt.isEmpty()) {
+            return;
+        }
+
+        TypeSymbol typeSymbol = ((ParameterSymbol) typeSymbolOpt.get()).typeDescriptor();
+        TypeDescKind actualTypeDesc = typeSymbol.typeKind();
+        if (actualTypeDesc != STRING) {
+            reportErrorDiagnostics(INVALID_PARAM_TYPE, requiredParam.typeName().location(),
+                    requiredParam.paramName().map(Node::toString).orElse(""), STRING.getName());
+        }
+    }
+
+    private void validateErrorParameter(ParameterNode parameterNode) {
+        if (!(parameterNode instanceof RequiredParameterNode requiredParam)) {
+            reportDiagnosticsForNonRequiredParam(parameterNode);
+            return;
+        }
+
+        Optional<Symbol> paramSymbolOpt = this.semanticModel.symbol(requiredParam);
+        if (paramSymbolOpt.isEmpty()) {
+            return;
+        }
+
+        TypeSymbol typeSymbol = ((ParameterSymbol) paramSymbolOpt.get()).typeDescriptor();
+        if (typeSymbol.typeKind() == TYPE_REFERENCE) {
+            if (!typeSymbol.getName().orElse("").equals(ERROR_PARAM) ||
+                    !isCdcModule(typeSymbol.getModule().orElse(null))) {
+                reportErrorDiagnostics(INVALID_PARAM_TYPE, parameterNode.location(),
+                        requiredParam.paramName().map(Node::toString).orElse(""),
+                        "error' or 'cdc:Error");
+            }
+        } else if (typeSymbol.typeKind() != ERROR) {
+            reportErrorDiagnostics(INVALID_PARAM_TYPE, parameterNode.location(),
+                    requiredParam.paramName().map(Node::toString).orElse(""),
+                    "error?' or 'cdc:Error?");
+        }
+    }
+
+    private void reportDiagnosticsForNonRequiredParam(ParameterNode parameterNode) {
+        Optional<Symbol> symbolOpt = semanticModel.symbol(parameterNode);
+        symbolOpt.ifPresent(symbol -> reportErrorDiagnostics(MUST_BE_REQUIRED_PARAM, parameterNode.location(),
+                symbol.getName().orElse("")));
```

Comment:
We can chain this together IMO.
```suggestion
        semanticModel.symbol(parameterNode)
            .ifPresent(symbol -> reportErrorDiagnostics(MUST_BE_REQUIRED_PARAM, parameterNode.location(),
                symbol.getName().orElse("")));
```

Reply:
Fixed in [c936ccb](https://github.com/ballerina-platform/module-ballerinax-cdc/pull/3/commits/c936ccbc4ffbe912c9b9ea28a7cf14acc07f0308)